lua/barbar/render.lua

```diff
@@ -152,9 +152,10 @@ local function slice_groups_right(groups, width)
     local text_width = strwidth(group.text)
     accumulated_width = accumulated_width + text_width
 
+    local ellipsis = '…'
     if accumulated_width >= width then
-      local diff = text_width - (accumulated_width - width)
-      local new_group = {hl = group.hl, text = strcharpart(group.text, 0, diff)}
+      local diff = text_width - (accumulated_width - width) - strwidth(ellipsis) + 2
```

Comment:
The length of `ellipsis` can be precalculated and stored in another constant (e.g. `ELLIPSIS_LEN`)

Reply:
In terms of the `+2` I I tried to exploratory test things like presence of borders and if the offset is obscured by multi-byte chars on truncation. It looked fine on what I tested.